solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/GatewayStateData.kt

```diff
@@ -0,0 +1,19 @@
+package org.p2p.solanaj.kits.renBridge
+
+import org.p2p.solanaj.core.AbstractData
+import org.p2p.solanaj.utils.ByteUtils
+import java.math.BigInteger
+
+private const val GATEWAY_STATE_DATA_LENGTH = 1 + 20 + 32 + ByteUtils.UINT_64_LENGTH + 1
+class GatewayStateData(private val data: ByteArray) : AbstractData(data, GATEWAY_STATE_DATA_LENGTH) {
+
+    val isInitialized: Boolean = readByte().toInt() != 0
+    val renVMAuthority: ByteArray = readBytes(20)
+    private val selectors: ByteArray = readBytes(32)
+    val burnCount: BigInteger = readUint64()
+    private val underlyingDecimals: Byte = readByte()
+
+    companion object {
+        fun decode(data: ByteArray) = GatewayStateData(data)
```

Comment:
```suggestion
        fun decode(data: ByteArray): GatewayStateData = GatewayStateData(data)
```

Reply:
when should i use companion object, and when constructor?

Reply:
I don't know, i just move it from nested class

Reply:
you can use any